### PR TITLE
#7717 Test Criteria isNull query

### PR DESCRIPTION
**Pull request by thicolares**
I wanted to demonstrate that I could not reproduce #7717 and it looks good.

### CODE REVIEW

**Comment by greg0ire** [comment]
Read [this slide](https://www.greg0ire.fr/git-gud/#/23) to learn how to avoid committing a .DS_Store file for life.

**Comment by thicolares** [comment]
omg. I've looked at the commit and haven't noticed. I'd set other things to my global gitignore (like .idea etc), but I think I hadn't problem with .DS_store in the past. Well, I'll add. Thank you!

**Comment by thicolares** [comment]
Fixed for good and rebased :)

**Review by greg0ire — DISMISSED** [review]

File: tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php [hunk 1]
```diff
@@ -105,6 +105,21 @@ public function testIsNullComparison()
...
+        $this->loadNullFieldFixtures();
+        $repository = $this->_em->getRepository(DateTimeModel::class);
+
+        $repository->matching(new Criteria(
+            Criteria::expr()->isNull('time')
+        ))->first();
+
+        $this->assertContains(
+            'col_time IS NULL',
+            $this->_sqlLoggerStack->queries[$this->_sqlLoggerStack->currentQuery]['sql']
```
Comment (SenseException) [inline tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php]: Checking for a part of a string in an SQL query is kind of a foggy test compared to the other tests in this class. I would prefer to work and compare with a result.

**Review by SenseException — CHANGES_REQUESTED** [review]

File: tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php (hunk 1 above)
Comment (thicolares) [reply tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php]: I see your point. But my intention was to demonstrate the query is created as expected given this criteria (due to #7717). What do you think?

Comment (SenseException) [reply tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php]: Despite that the query part was created I like the test to show that `isNull` is also returning the expected result when the query gets executed. Checking the SQL itself isn't the same as having the expected result of the query. This way regressions in the functionality can be covered too for any other unwanted changes that might be introduced besides `'col_time IS NULL'`.

**Pull request closed without merging**